File: ProjectPos.BackOffice.FrontEnd/src/proxy/interfaces/proof-of-payment-report-dto.ts
import { ProofOfPaymentDto } from "./proof-of-payment-dto";

export interface ProofOfPaymentReportDto {
    userId?: number;
    userName?: string;
    customerId?: number;
    customerName?: string;
    createdDate?: Date;
    bookingDate?: Date;
    proofOfPayments?: ProofOfPaymentDto[];
    paidTotal?: number;
    usableTotal?: number;
    usedTotal?: number;
}

File: ProjectPos.BackEnd/ProjectPos.Services/ReportingDtos/ProofOfPaymentReportDto.cs
using ProjectPos.Services.DTOs;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProjectPos.Services.ReportingDtos
{
    public class ProofOfPaymentReportDto
    {
        public int? UserId { get; set; }
        public string? UserName { get; set; }
        public int? CustomerId { get; set; }
        public string? CustomerName { get; set; }
        public DateTime? CreatedDate { get; set; }
        public DateTime? BookingDate { get; set; }
        public IEnumerable<ProofOfPaymentDto>? ProofOfPayments { get; set; }
        public decimal? PaidTotal { get; set; }
        public decimal? UsableTotal { get; set; }
        public decimal? UsedTotal { get; set; }
    }
}
